src/Doctor/Doctor.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Dock\Doctor;
+
+class Doctor
+{
+    /**
+     * @var array
+     */
+    private $tasks;
+
+    /**
+     * @param array $tasks
+     */
+    public function __construct(array $tasks)
+    {
+        $this->tasks = $tasks;
+    }
+
+    public function examine($dryRun)
+    {
+        foreach ($this->tasks as $task) {
+            $task->run($dryRun);
```

Comment:
`Task::run` returns null. Currently, failed tests throw exceptions and interrupt execution. I'm not sure that's a good solution, but mostly we might expect these tests are ordered and the subsequent tests rely on the previous ones passing.